lib/archive_importer.rb

```diff
@@ -37,12 +37,26 @@ def create_user(attr)
     )
     self.user = User.build(data)
     user.save!
+    import_blocks(attr, user)
   end
 
   private
 
   attr_reader :archive_hash
 
+  def import_blocks(attr, user)
+    return unless attr[:blocks]
+
+    attr[:blocks].each do |blocked_id|
+      p = Person.find_or_fetch_by_identifier(blocked_id)
```

Comment:
This raises an error when the person isn't known and can't be fetched, which would abort the whole import. Also, if the blocked account was migrated between the export-creation and the import, you want to block the new user, not the old one. So in the end you probably want a similar logic for blocks as it already exists for contacts here: https://github.com/diaspora/diaspora/blob/develop/lib/archive_validator/contact_validator.rb

Reply:
Do you mean the line: ` person.account_migration.newest_person` to fetch migrated user? 

The 'find_or_fetch_by_identifier' method will not throw a exception any more if person can not be fetched, (PR #8257 will introduce this). Many methods rely on returning nil if person is not found, so I globally made this even for callers. 
See:  https://github.com/diaspora/diaspora/blob/d27eefeb3471844179c86c8dc7c28ad99801f40f/lib/archive_validator/contact_validator.rb#L34

Until 8257 a Person-fetch might never return nil if a person is not found.